### ISSUE TITLE
curl: (6) Could not resolve host: launchermeta.mojang.com

### ISSUE THREAD

**Issue by M4TY** [itzg/docker-minecraft-server#818]
Hello, when creating your Docker Container in Container Station on my QNAP NAS and I choose Bridge mode with custom local IP, I always get this error:

**_curl: (6) Could not resolve host: launchermeta.mojang.com_** 

This doesn't happen when I choose NAT mode in network settings. I read through many issues in this repo, but none of them helped me. I even tried _SKIP_NSSWITCH_CONF_ which doesn't help either. Thanks in advance.

**Comment by itzg** [comment]
I'm not aware of anything I can do to address this. I would prefer to close this, but welcome any others to find and add to this issue as more is known.

**Comment by pahansen95** [comment]
I ran into the same curl error but was running in an instance of microK8s on Ubuntu 20.04 on an Intel NUC.

After reading through #756 I spun up a container from the `adoptopenjdk/openjdk11:alpine-jre` image & found that I couldn't resolve any DNS names.

After a bunch of googling I found [this Stack Overflow post](https://stackoverflow.com/a/65593511) which summarizes what I found.

TL;DR It's caused by Alpine & the musl library not playing nice with Kubernetes ndots (see [resolv.conf](https://www.man7.org/linux/man-pages/man5/resolv.conf.5.html)) strategy. Probably also has something to do with my network's specific DNS environment.

Switching to the `multiarch` tag which is based on Debian instead of Alpine fixed my issues.

**Comment by itzg** [comment]
@pahansen95 Glad to hear the Debian based image resolved the issue for you. As a counter-datapoint I run primarily on microk8s on Ubuntu 20.04 and the alpine based image works fine for me; however, I'm using standard DNS provided by my ASUS router.

**Comment by pahansen95** [comment]
That's good to know @itzg. Do you have the coreDNS Addon enabled for microK8s?

I'm think it's specifically my microk8s environment. I don't have any issues if I spin up a `adoptopenjdk/openjdk11:alpine-jre` container using docker on the host.

@M4TY Does switching to the `multiarch` image resolve your issue as well?

**Comment by itzg** [comment]
> Do you have the coreDNS Addon enabled for microK8s?

I double checked and yes, I do.

**Comment by MegaShinySnivy** [comment]
Please excuse my late post, but I am having the exact same issue on K8s (Debian host) with both the Deb and Alpine-based images. The image was spun up in Rancher.

**Comment by drewp** [comment]
I had related troubles, and worked around my bad default k3s pod config like this (10.2.0.1 is my non-cluster dns):

```
apiVersion: apps/v1
kind: Deployment
spec:
  replicas: 1
  template:
    spec:
      dnsPolicy: None
      dnsConfig: {nameservers: [10.2.0.1]}
      containers:
        - name: minecraft
          image: itzg/minecraft-server:2021.17.0-java16-openj9
etc
```

**Comment by johnynfulleffect** [comment]
Same issue @MegaShinySnivy I tried both multiarch and latest.  Using Rancher k3s

**Comment by johnynfulleffect** [comment]
@drewp 's solution worked for me, but this option is not in the helm chart, so not ideal
